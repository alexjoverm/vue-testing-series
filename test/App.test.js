import { shallowMount } from '@vue/test-utils'
import App from '../src/App'

describe('App.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallowMount(App)
    cmp.setData({ messages: ['Cat'] });
  })

  it('equals messages to ["Cat"]', () => {
    expect(cmp.vm.messages).toEqual(['Cat'])
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})

/**
 * Without vue-test-utils
 */
// import Vue from 'vue'
// import App from '../src/App'

// describe('App.test.js', () => {
//   let cmp, vm

//   beforeEach(() => {
//     cmp = Vue.extend(App)
//     vm = new cmp({
//       data: {
//         messages: ['Cat']
//       }
//     }).$mount()
//   })

//   it('equals messages to ["Cat"]', () => {
//     expect(vm.messages).toEqual(['Cat'])
//   })

//   it('has the expected html structure', () => {
//     expect(vm.$el).toMatchSnapshot()
//   })
// })
