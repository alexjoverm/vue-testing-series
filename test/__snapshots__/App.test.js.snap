// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <messagelist-stub
    messages="Cat"
  />
</div>
`;
